### PR TITLE
#269 rework #248

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang@uk.ibm.com>

### CODE REVIEW

**Comment by Emily-Jiang** [comment]
@jmesnil please review. Please let me know whether the tcks run ok against your impl. thanks

File: spec/src/main/asciidoc/converters.asciidoc [hunk 1]
```diff
@@ -65,9 +65,5 @@ A custom `Converter` for a target type of any of the built-in Converters will ov
...
 
 * The target type `T` has a Constructor with a String parameter, or
-* the target type `T` has a Constructor with a CharSequence parameter, or
-* the target type `T` has a `static T valueOf(String)` method, or
-* the target type `T` has a `static T valueOf(CharSequence)` method, or
-* the target type `T` has a `static T parse(String)` method, or
-* the target type `T` has a `static T parse(CharSequence)` method, or
-* the target type `T` is an `Enum`, in which case the `Enum#valueOf(Class, String)` is being used.
+* the target type `T` has a `static T valueOf(String)` method
+If the target type `T` is an `Enum`, in which case the `Enum#valueOf(Class, String)` is being used.
```
Comment (jmesnil) [inline spec/src/main/asciidoc/converters.asciidoc]: This is not tested by the TCK

File: api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java [hunk 2]
```diff
@@ -69,12 +69,7 @@
  * the {@code Config} provides an <em>Implicit Converter</em>, if</p>
  * <ul>
  *     <li>The target type {@code T} has a Constructor with a String parameter, or</li>
- *     <li>The target type {@code T} has a Constructor with a CharSequence parameter, or</li>
  *     <li>the target type {@code T} has a {@code static T valueOf(String)} method, or</li>
- *     <li>the target type {@code T} has a {@code static T valueOf(CharSequence)} method, or</li>
- *     <li>the target type {@code T} has a {@code static T parse(String)} method, or</li>
- *     <li>the target type {@code T} has a {@code static T parse(CharSequence)} method, or</li>
- *     <li>the target type {@code T} is an {@code Enum}, in which case the {@link Enum#valueOf(Class, String)} is being used.</li>
  * </ul>
```
Comment (jmesnil) [inline api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java]: The line about Enum should be kept

File: spec/src/main/asciidoc/converters.asciidoc (hunk 1 above)
Comment (jmesnil) [reply spec/src/main/asciidoc/converters.asciidoc]: I opend #278 to test implicit Enum conversion

File: api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java (hunk 2 above)
Comment (Emily-Jiang) [reply api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java]: We discussed this and John suggested to remove it as it is covered by the valueOf. @johnament commented: Technically, this last case isn't needed.  When an Enum type is created, it automatically gets a T.valueOf method.
It is implementation detail. The enum is covered under valueOf.

Comment (jmesnil) [reply api/src/main/java/org/eclipse/microprofile/config/spi/Converter.java]: ok, that's fine.

**Pull request merged**